### ISSUE TITLE
aws-cdk-lib/aws-dynamodb: Enable SSE

### ISSUE THREAD

**Issue by jpmatusik** [aws/aws-cdk#15004]
I don't see SSE being enabled when I make a DynamoDB table.

I used 

```
    new dynamodb.Table(this, 'MyTable', {
      partitionKey: { name: "someKey", type: dynamodb.AttributeType.STRING},
      encryption: dynamodb.TableEncryption.DEFAULT
    });

```

### Reproduction Steps

```
# pull minimal example
git clone https://github.com/jpmatusik/aws-cdk-dynamodb-bug.git
cd aws-cdk-dynamodb-bug

# build docker image with nodejs, ts, aws-cdk
docker build -t dynamodb-bug .
docker run -it -v $PWD:/dynamodb-bug dynamodb-bug

# from inside the container
cd /dynamodb-bug
npm install # takes forever 🤷 
npm run build && npx jest # takes a bit less than forever 🤷 
```

### What did you expect to happen?

I want this to show up in the table's cloudformation attributes.
```
{
  SSESpecification: {
    SSEEnabled: true
  }
}
```

Build a CloudFormation document containing a DynamoDB table with SSE.

### What actually happened?

The DynamoDB table does not have SSE.

### Environment

  - CDK CLI Version: 2.0.0-rc.6 (build 849f0c0)
  - Node.js Version: v14.17.0
  - OS: macOS Big Sur 11.2.3 
  - Docker Image: node:14.17-slim
  - typescript: Version 4.3.2

--- 

### Other

This is the output of the test...

```
    None of 1 resources matches resource 'AWS::DynamoDB::Table' with {
      "$objectLike": {
        "SSESpecification": {
          "SSEEnabled": true
        }
      }
    }.
    - Field SSESpecification mismatch: Field SSEEnabled mismatch: Different values in:
        {
          "Type": "AWS::DynamoDB::Table",
          "Properties": {
            "KeySchema": [
              {
                "AttributeName": "someKey",
                "KeyType": "HASH"
              }
            ],
            "AttributeDefinitions": [
              {
                "AttributeName": "someKey",
                "AttributeType": "S"
              }
            ],
            "ProvisionedThroughput": {
              "ReadCapacityUnits": 5,
              "WriteCapacityUnits": 5
            },
            "SSESpecification": {
              "SSEEnabled": false <- Want this guy to be true!
            }
          },
          "UpdateReplacePolicy": "Retain",
          "DeletionPolicy": "Retain"
        }

```

This is :bug: Bug Report

**Comment by cjihrig** [comment]
It looks like you're using `dynamodb.TableEncryption.DEFAULT`, which maps to ["Server-side KMS encryption with a master key owned by AWS."](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-dynamodb.TableEncryption.html)

According to the [`SSEEnabled` CFN documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-ssespecification.html):

> If disabled (false) or not specified, server-side encryption is set to AWS owned CMK.

It sounds like what you're seeing is expected behavior. If you want `SSEEnabled` to be `true`, I think you'd want to use `dynamodb.TableEncryption.AWS_MANAGED`.

**Comment by jpmatusik** [comment]
I see. Yeah from the docs you pointed that makes sense. I was working off the docs here:
https://cdkworkshop.com/20-typescript/70-advanced-topics/100-construct-testing/1000-assertion-test.html

Those docs are for the 1.X version though so I had to make a change. In the 1.X version, `serverSideEncryption` is explicitly set. So when I used the `encryption` attribute in the 2.X version, I thought any of TableEncryption enum values would set SSEEnabled to true. I should have looked a little deeper. Thanks!
